pgsqltoolsservice/driver/driver_manager.py

```diff
@@ -0,0 +1,34 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from abc import ABC, abstractmethod
+from pgsqltoolsservice.parsers.owner_uri_parser import get_attribute_value
+from pgsqltoolsservice.driver.types import *
+
+class DriverManager:
+    """Wrapper class that handles different types of drivers and connections """
+
+    def __init__(self, provider, params):
+
+        # Get info about this connection's provider
+        self._provider = provider
+
+        # Create a connection using the provider and connection options
+        self._conn_object = self._create_connection(params.connection.options)
+
+    def _create_connection(self, options) -> ServerConnection:
+        """
+        Creates a ServerConnection according to the provider and connection options
+        :param options: a dict containing connection parameters
+        """
+        if self._provider == "PGSQL":
+            return PsycopgConnection(options)
```

Comment:
I am thinking whether we have to name the connection objects in terms of driver or just PostgreSQLConnection and MySQLConnection ? What if we want to change the underlying driver tomorrow ? We will have to change all the class names.

Reply:
What I was thinking was, if we want to use a different driver for PGSQL for example, we can simply create a new class with that driver that implements ServerConnection, and change line 27 in ConnectionManager to point to our new class. This allows us to switch between different drivers for the same provider by only changing this line. However, if that isn't something we want, I can rename the PsycopgConnection and PyMySQLConnection classes as you suggested.